### PR TITLE
Feature/helm documentation

### PR DESCRIPTION
**Pull request by bvaughan-nexient**
Helm Standards and Practices documentation.

* Deployment chart standards
* Library chart standards
* Helper template standards

Continuous Deployment

* Continuous Deployment as application architecture feature
* Evolutionary Database as application architecture feature
* Helm as component of CD on Kubernetes clusters

### CODE REVIEW

File: standards/helm-development/charts/deployment/README.md [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+  - [Chart Structure](#chart-structure)
+  - [Values file](#values-file)
+  - [Variable Naming](#variable-naming)
+  - [Templates and Helpers](#templates-and-helpers)
+  - [Dependencies](#dependencies)
+  - [Testing and Validation](#testing-and-validation)
+  - [Documentation](#documentation)
+
+
+## Chart Naming
```
Comment (chris11-taylor-nttd) [inline standards/helm-development/charts/deployment/README.md:24]: We've got some repo naming standards laid down [here](https://github.com/nexient-llc/common-platform-documentation/blob/main/standards/common-development/git/repository/naming-schemes/iac-repository-names.md), how does this factor in?

File: platform/continuous-deployment/immutable-application-deployment/README.md [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+**Containerization:** Use containerization technologies to package your application and its dependencies into a single, immutable unit.
+
+**Configuration Management:** Store configuration separately from your application, such as in environment variables or external configuration services. This allows you to change the configuration without changing the application itself. These configuration sets should be versionable with the ability to correlate a configuration version with a deployment or application version.
+
+**Automate Deployments:** Use automation tools to automate the process of deploying your application. This reduces the risk of human error and ensures that every deployment is consistent.
+
+**Version Control:** Keep track of all versions of your application. This allows you to quickly roll back to a previous version if something goes wrong.
+
+**Health Checks:** Implement health checks at all levels of the application to monitor your application. This allows the system to perform some level of self-healing and allows SRE and incident response resources to have a better understanding of the state of the application should an incident arise.
```
[the remark this answers is not in the document]
Comment (bvaughan-nexient) [reply platform/continuous-deployment/immutable-application-deployment/README.md:35]: Good call out -- I may just use your text.

File: standards/helm-development/charts/deployment/README.md (hunk 1 above)
Comment (bvaughan-nexient) [reply standards/helm-development/charts/deployment/README.md:24]: I went back and forth on this. I don't think the repo naming applies 100%, but the chart naming should be familiar to someone who knows the repo naming standards. I will take another look to be sure this is true.

File: platform/continuous-deployment/evolutionary-database-deployment/README.md [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+## Conclusion
+
+Evolutionary database migration is a powerful technique that enables continuous deployment of applications by allowing for seamless updates to the database schema. By following a structured migration process and leveraging automated tools, developers can make incremental changes to the database while preserving existing data and ensuring the application's functionality. This approach promotes collaboration, flexibility, and reliability in managing database changes, ultimately leading to faster and more efficient application deployment.
+
+---
+**Document Revision History**
+| Date | Version | Author |Notes |
+| --- | --- | --- | --- |
+| 2024-03-15 | 1.0 | Ben Vaughan | Initial Release |
```
[the remark this answers is not in the document]
Comment (bvaughan-nexient) [reply platform/continuous-deployment/evolutionary-database-deployment/README.md:41]: I'll go with using the git history. Most of my experience is with Wiki's (Confluence) where revision history was present, but not as useful as one would hope.

I'll drop it.

**Review by chris11-taylor-nttd — APPROVED** [review]

**Review by dsahoo-nexient — APPROVED** [review]

**Pull request merged**